File: app/src/main/java/jp/juggler/subwaytooter/api/entity/TootGap.java
package jp.juggler.subwaytooter.api.entity;

public class TootGap {
	
	public final String max_id;
	public final String since_id;
	
	public TootGap( String max_id, String since_id ){
		this.max_id = max_id;
		this.since_id = since_id;
	}
	
	//	public TootGap( long max_id, String since_id ){
	//		this.max_id = Long.toString(max_id);
	//		this.since_id = since_id;
	//	}
	
	public TootGap( long max_id, long since_id ){
		this.max_id = Long.toString( max_id );
		this.since_id = Long.toString( since_id );
	}
}

File: app/src/main/java/jp/juggler/subwaytooter/api/entity/TootGap.kt
package jp.juggler.subwaytooter.api.entity

class TootGap(
    var maxId: EntityId?, // TLの先頭にギャップを置くかもしれない
    val sinceId: EntityId,
) : TimelineItem() {

    companion object {
        fun mayNull(maxId: EntityId?, sinceId: EntityId?) =
            sinceId?.let { TootGap(maxId, it) }
    }

    override fun getOrderId(): EntityId = sinceId

    //	constructor(max_id : Long, since_id : Long) : this(EntityIdLong(max_id), EntityIdLong(since_id))
}
